src/Symfony/Component/Intl/DateFormatter/IntlDateFormatter.php

```diff
@@ -210,14 +210,16 @@ public function format($timestamp)
         // behave like the intl extension
         $argumentError = null;
         if (!is_int($timestamp)) {
-            if (PHP_VERSION_ID < 50304) {
-                $argumentError = 'datefmt_format: takes either an array  or an integer timestamp value ';
-            } elseif (!$timestamp instanceof \DateTime) {
-                $argumentError = 'datefmt_format: takes either an array or an integer timestamp value or a DateTime object';
+            if (!$timestamp instanceof \DateTime) {
                 if (PHP_VERSION_ID >= 50500 || (extension_loaded('intl') && method_exists('IntlDateFormatter', 'setTimeZone'))) {
                     $argumentError = sprintf('datefmt_format: string \'%s\' is not numeric, which would be required for it to be a valid date', $timestamp);
+                } elseif (PHP_VERSION_ID >= 50304) {
+                    $argumentError = 'datefmt_format: takes either an array or an integer timestamp value or a DateTime object';
                 }
             }
+            if (PHP_VERSION_ID < 50304 && null === $argumentError) {
+                $argumentError = 'datefmt_format: takes either an array  or an integer timestamp value ';
```

Comment:
just verified, this was already there before, and matches php 5.3.3 error message...